src/clp_ffi_js/ir/StructuredIrUnitHandler.cpp

```diff
@@ -0,0 +1,145 @@
+#include "StructuredIrUnitHandler.hpp"
+
+#include <algorithm>
+#include <cctype>
+#include <cstddef>
+#include <iterator>
+#include <memory>
+#include <string>
+#include <string_view>
+#include <type_utils.hpp>
+#include <utility>
+
+#include <clp/ffi/ir_stream/decoding_methods.hpp>
+#include <clp/ffi/KeyValuePairLogEvent.hpp>
+#include <clp/ffi/SchemaTree.hpp>
+#include <clp/ffi/Value.hpp>
+#include <emscripten/val.h>
+#include <spdlog/spdlog.h>
+
+#include <clp_ffi_js/constants.hpp>
+#include <clp_ffi_js/ir/LogEventWithFilterData.hpp>
+
+namespace clp_ffi_js::ir {
+
+namespace {
+/**
+ * Parses the `LogLevel` from an input string.
+ * @param str
+ * @return
+ */
+auto parse_log_level(std::string_view str) -> LogLevel;
+
+auto parse_log_level(std::string_view str) -> LogLevel {
+    LogLevel log_level{LogLevel::NONE};
+
+    // Convert the string to uppercase.
+    std::string log_level_name_upper_case{str};
+    std::ranges::transform(
+            log_level_name_upper_case.begin(),
+            log_level_name_upper_case.end(),
+            log_level_name_upper_case.begin(),
+            [](unsigned char c) { return std::toupper(c); }
+    );
+
+    auto const* it = std::ranges::find(
+            cLogLevelNames.begin() + clp::enum_to_underlying_type(cValidLogLevelsBeginIdx),
+            cLogLevelNames.end(),
+            log_level_name_upper_case
+    );
+
+    if (it == cLogLevelNames.end()) {
+        return log_level;
+    }
```

Comment:
Would this be easier to read?
```suggestion
    if (it == cLogLevelNames.end()) {
        return LogLevel::NONE;
    }
```
And below we can either return the casted log level inline or wrap it with a declaration then return.